workers.py

```diff
@@ -44,6 +44,100 @@ def _retryable(method: str, *args, **kwargs) -> requests.Response:
     raise requests.HTTPError('All attempts failed')
 
 
+def query_main_collection(
+        host: str,
+        endpoint: str,
+        main_collection: str,
+        query_string: str
+) -> dict:
+    """Query a Collection.
+
+    Returns a Response object with data retrieved from all pages
+    :param host: The host
+    :param endpoint: API Endpoint
+    :param main_collection: Collection name ('container_nodes', 'sources' etc)
+    :param query_string: Query string as additional params to the GET request
+    :return: Response object with data retrieved from all pages
+    :raises: HTTPError in the function caller
+    """
+    # First GET call to get data as well as pagination links
+    resp = _retryable(
+        'get',
+        f'{host}{endpoint}/{main_collection}',
+        params={query_string: ''},
+        verify=False
+    )
+    out = resp.json()
+    all_data = out['data']
+    prometheus_metrics.METRICS['get_successes'].inc()
+
+    # Subsequent GET calls that reference the pagination link
+    while out['links'].get('next'):
```

Comment:
The code can be a bit simpler if you'd abstract the crawl of the "next" pages to a separate generator/function. So it can be stripped away from the second function and made generic.

Reply:
Extracted this out in `aggregated_pagination_data` method so both functions `query_main_collection` and `query_sub_collection` could use it